networking/external_dns_operator/nw-control-dns-records-public-hosted-zone-gcp.adoc

```diff
@@ -0,0 +1,151 @@
+[id="nw-control-dns-records-public-hosted-zone-gcp_{context}"]
+= Creating DNS records on an public managed zone for GCP by using Red Hat External DNS Operato
+
+.Procedure
+
+. Export the kubeconfig file.
++
+[source,terminal]
+----
+$ export KUBECONFIG=~/<path to>/kubeconfig
+----
+
+. Check the user. The user must have access to the `kube-system` namespace if you don’t have the credentials as you can fetch the credentials from the `kube-system` namespace to use the cloud provider client.
++
+[source,terminal]
+----
+$ oc whoami
+----
++
+.Example output
+[source,terminal]
+----
+system:admin
+----
+
+. Copy the value of service_account.json in gcp-credentials secret in a file encoded-gcloud.json
++
+[source,terminal]
+----
+$ oc get secret gcp-credentials -n kube-system --template='{{$v := index .data "service_account.json"}}{{$v}}' | base64 -d - > decoded-gcloud.json
+----
+
+. Export Google credentials:
++
+[source,terminal]
+----
+$ export GOOGLE_CREDENTIALS=decoded-gcloud.json
+----
+
+. Activate your account:
++
+[source,terminal]
+----
+$ gcloud auth activate-service-account  <client_email as per decoded-gcloud.json> --key-file=decoded-gcloud.json
+----
+
+. Set your project
++
+[source,terminal]
+----
+$ gcloud config set project <project_id as per decoded-gcloud.json>
+----
+
+. Get the routes to check the domain
++
+[source,terminal]
+----
+$ oc get routes --all-namespaces | grep console
+----
++
+.Example output
+[source,terminal]
+----
+openshift-console          console             console-openshift-console.apps.qe.gcp.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.qe.gcp.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+----
+
+. Get your zone which was created by the installer
++
+[source,terminal]
+----
+$ gcloud dns managed-zones list | grep qe.gcp.devcluster.openshift.com
+
+qe-cvs4g-private-zone qe.gcp.devcluster.openshift.com
+----
+
+. Check the zone doesn't have DNS records other than `NS` and `SOA`
++
+[source,terminal]
+----
+$ gcloud dns record-sets list --zone=qe-cvs4g-private-zone
+----
+
+. Create a ExternalDNS CR
++
+[source,yaml]
+----
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-gcp <1>
+spec:
+  domains:
+    - filterType: Include <2>
+      matchType: Exact <3>
+      name: qe.gcp.devcluster.openshift.com <4>
+  zones:
+    - qe.gcp.devcluster.openshift.com <5>
```

Comment:
As per [this comment](https://github.com/openshift/openshift-docs/pull/40441/files#r809474595), to avoid the redundant examples I propose to drop `zones` for GCP as they can be either IDs or zone names, none of this shows the correlation to the routes' hostnames.
The points 2,3,4 can be reworded the same way.

Reply:
ack